internal/updater/checker.go

```diff
@@ -19,26 +19,36 @@ type httpGetter interface {
 	Get(string) ([]byte, error)
 }
 
+type Configurable interface {
+	GetString(string) string
+}
+
 type Checker struct {
-	apiURL         string
+	cfg            Configurable
+	apiInfoURL     string
+	fileURL        string
 	currentChannel string
 	currentVersion string
 	httpreq        httpGetter
 }
 
-var DefaultChecker = newDefaultChecker()
-
-func newDefaultChecker() *Checker {
+func DefaultChecker(cfg Configurable) *Checker {
```

Comment:
Suggest naming this `NewDefaultChecker`, or something else that clearly conveys we're creating a new checker rather than using some existing default.